Sources/App/Models/Reference.swift

```diff
@@ -47,6 +47,11 @@ extension Reference: Codable {
             self = value.asReference
             return
         }
+        // try legacy format last
+        if let value = try? container.decode(SemVer.self, forKey: .tag) {
+            self = .tag(value, nil)
+            return
+        }
```

Comment:
I'm actually not convinced we should try and recover from this given that we're going to squash the scheme and clear out that bad data. It's going to be cruft immediately after that.

I think the main lesson learned is to be careful (and a test) when making changes to our JSON payloads. What happened here was that when you go from one field to two the JSON format changes from an inline field to an object. And that's not something you can simply recover from by adding an optional field.

I think I'd just revert that last commit, only add the error reporting, and then let the schema squash handle the rest.

What do you think?

Reply:
PS: This failure to clear out wasn't a widespread issue, because we deployed the initial format change a while back. Most fields were processed fine.

The issue where the format was a problem was only introduced when I added scoring. At that point `updateStatus` started to fail if it encountered records with old `Reference` data. And there were a few holdouts that haden't been processed yet, presumably because they'd failed for other reasons until now.

Reply:
> What do you think?

I'm fine with what you say here, but I think it's worth talking about something else around the same subject. I've noticed a marked reduction in Rollbar errors (actually, to zero over the last several days). This kind of crash is exactly what I'd like Rollbar to notify us of. Someone hits a page, and it 500s while generating it.

Should that be happening already? If not, I think we should try to make it happen.

Reply:
Yes, there's really only one critical error we're currently reporting: if the checkouts directory can't be created.

I agree, this here is a worthy candidate :D I'll roll back the field change and add reporting a critical error up to rollbar.

Reply:
Can we make more generic, so all 500s during a page render result in a Rollbar error?

Reply:
I'm sure we can hook that into the middleware. I'll add an issue to track, ok?

Reply:
Basically, if someone expected to see a page and instead saw an error, I'd like to at least be aware.

Errors like this are super important. If someone tries the site for the first time and see an error when clicking through to a package, they're going to judge the whole site by that one result, and potentially never come back. It could be failing for days/weeks on some packages if we don't know about this kind of error.

Reply:
> I'm sure we can hook that into the middleware. I'll add an issue to track, ok?

Perfect, thanks!

Reply:
I also just bumped up the priority of #125 into Essential for the same reason. That raw JSON error is a killer to people's confidence in the site. We need to make it render errors in the page template.

Reply:
Yep, totally!